### PR TITLE
Replace identical files with CoreLib mirror

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
contributes to #27841

Thank's to @EgorBo i removed 100% identical files from Common folder with CoreLib mirror.

---RESOLVED---
Anyway there is one file left out: 

`corefx\src\Common\src\System\Threading\Tasks\TaskToApm.cs` in `corefx\src\System.Net.Http\src\System.Net.Http.csproj`

if i try to substitute with 
``` c#
<Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
     <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
</Compile>
```

i get this error when compile:
``` 
CSC : error CS2001: Source file '...\corefx\src\Common\src\System\Threading\Tasks\TaskToApm.cs' could not be found. [...\corefx\src\System.Net.Http\src\System.Net.Http.csproj]
``` 
@karelz what i'm missing?
---RESOLVED---

cc: @danmosemsft 

Tested on Win/Ubuntu

### CODE REVIEW

**Comment by danmosemsft** [comment]
The path in the error above does not contain corelib so I don't see how it could be produced from the project change you pasted.

**Comment by MarcoRossignoli** [comment]
@danmosemsft [here](https://github.com/dotnet/corefx/commit/ba25b6f8ef1565ea1b908bdfc08d2a6d397a8cdd) a branch(dup2) with full changes...do you compile well?Also to me this is very strange...all others are ok.

**Comment by danmosemsft** [comment]
@MarcoRossignoli we are stabilizing for 2.1 now: we want to hit zero bugs tomorrow. Because this change is cleanup rather than functionally important I would prefer to hold off merging until master becomes a 2.2 branch, in a small number of weeks.

I certainly want to merge this cleanup then though.

**Comment by MarcoRossignoli** [comment]
@ViktorHofer as @danmosemsft said, we can wait "2.2 branch" and i'll check if you PR will be merged. If merged i'll rebase/re-run custom task. What do you think?

**Comment by ViktorHofer** [comment]
Great, thanks!

**Comment by MarcoRossignoli** [comment]
@danmosemsft i launch a find to understand if i missed some place: 
`findstr /n /s /c:"TaskToApm.cs"  *.csproj`
result:
```
src\Common\tests\Common.Tests.csproj:96:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\Common\tests\Common.Tests.csproj:97:      <Link>ProductionCode\Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Data.SqlClient\src\System.Data.SqlClient.csproj:202:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Data.SqlClient\src\System.Data.SqlClient.csproj:203:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.IO.Compression\src\System.IO.Compression.csproj:44:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.IO.Compression\src\System.IO.Compression.csproj:45:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.IO.Compression\tests\System.IO.Compression.Tests.csproj:57:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.IO.Compression\tests\System.IO.Compression.Tests.csproj:58:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.IO.Compression.Brotli\src\System.IO.Compression.Brotli.csproj:27:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.IO.Compression.Brotli\src\System.IO.Compression.Brotli.csproj:28:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.IO.Compression.Brotli\tests\System.IO.Compression.Brotli.Tests.csproj:31:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.IO.Compression.Brotli\tests\System.IO.Compression.Brotli.Tests.csproj:32:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.IO.Pipes\src\System.IO.Pipes.csproj:34:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.IO.Pipes\src\System.IO.Pipes.csproj:35:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Net.Http\src\System.Net.Http.csproj:483:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Net.Http\src\System.Net.Http.csproj:484:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Net.Http\src\System.Net.Http.csproj:585:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Net.Http\src\System.Net.Http.csproj:586:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Net.Http\tests\UnitTests\System.Net.Http.Unit.Tests.csproj:84:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Net.Http\tests\UnitTests\System.Net.Http.Unit.Tests.csproj:85:      <Link>ProductionCode\Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Net.Http.WinHttpHandler\tests\UnitTests\System.Net.Http.WinHttpHandler.Unit.Tests.csproj:88:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Net.Http.WinHttpHandler\tests\UnitTests\System.Net.Http.WinHttpHandler.Unit.Tests.csproj:89:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Net.HttpListener\src\System.Net.HttpListener.csproj:362:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Net.HttpListener\src\System.Net.HttpListener.csproj:363:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Net.NetworkInformation\src\System.Net.NetworkInformation.csproj:264:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Net.NetworkInformation\src\System.Net.NetworkInformation.csproj:265:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Net.Requests\src\System.Net.Requests.csproj:95:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Net.Requests\src\System.Net.Requests.csproj:96:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Net.Security\src\System.Net.Security.csproj:96:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Net.Security\src\System.Net.Security.csproj:97:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Net.Security\tests\FunctionalTests\System.Net.Security.Tests.csproj:87:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Net.Security\tests\FunctionalTests\System.Net.Security.Tests.csproj:88:      <Link>ProductionCode\Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Net.Sockets\src\System.Net.Sockets.csproj:291:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Net.Sockets\src\System.Net.Sockets.csproj:292:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Net.Sockets\tests\FunctionalTests\System.Net.Sockets.Tests.csproj:104:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Net.Sockets\tests\FunctionalTests\System.Net.Sockets.Tests.csproj:105:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Runtime.Extensions\src\System.Runtime.Extensions.csproj:73:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Runtime.Extensions\src\System.Runtime.Extensions.csproj:74:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
src\System.Security.Cryptography.Primitives\src\System.Security.Cryptography.Primitives.csproj:33:    <Compile Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs">
src\System.Security.Cryptography.Primitives\src\System.Security.Cryptography.Primitives.csproj:34:      <Link>Common\CoreLib\System\Threading\Tasks\TaskToApm.cs</Link>
```
i don't see any place with `Common\src\System\Threading\Tasks\TaskToApm.cs`
but if i add verbose log to msbuild(/v:d) i found that in list of `CoreCompile`
```
Project "...\corefx\src\System.Net.Http\src\System.Net.Http.csproj" on node 1 (default targets).
CoreResGen:
  No resources are out of date with respect to their source files. Skipping resource generation.
GenerateResourcesSource:
Skipping target "GenerateResourcesSource" because all output files are up-to-date with respect to the input files.
GenerateAssemblyInfo:
Skipping target "GenerateAssemblyInfo" because all output files are up-to-date with respect to the input files.
CoreCompile:
...corefx\src\Common\src\System\Threading\Tasks\TaskToApm.cs...
```
Is it possible build process generate that "command" on-the-fly(something out of csproj) for some reason?

**Comment by danmosemsft** [comment]
@maryamariyan can you please help @MarcoRossignoli  with this one? I think you can clone the branch.

**Comment by MarcoRossignoli** [comment]
@maryamariyan you can clone [this](https://github.com/MarcoRossignoli/corefx/tree/dups2) and run simple build...you should get compile error, TYVM for help.

**Comment by maryamariyan** [comment]
@danmosemsft @MarcoRossignoli sure, looking...

**Comment by MarcoRossignoli** [comment]
>update
I was able to reproduce. I'll need to spend more time on this. will comment back when I find a solution.

@maryamariyan if i can help to investigate in some "direction" let me know.

**Comment by maryamariyan** [comment]
@MarcoRossignoli you will need to remove [this line](https://github.com/dotnet/corefx/blob/master/src/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.msbuild#L30) to fix the compile issue present in the dups branch

**Comment by MarcoRossignoli** [comment]
@maryamariyan TYVM i did a search only in *.csproj :triumph:(i have to learn build process for this repo, if you know if there are some docs around left me know)

>I would prefer to hold off merging until master becomes a 2.2 branch, in a small number of weeks.

@danmosemsft is there a date(or doc to follow to know when 2.2 branch will start)?

**Comment by maryamariyan** [comment]
@MarcoRossignoli I used the /pp option.
>msbuild path_to_file.csproj /pp

You can find some documentation [here](https://github.com/Microsoft/msbuild/blob/b657647d2e6f4ed20ce6cb3201a55ee02f09c062/documentation/wiki/MSBuild-Tips-%26-Tricks.md)

**Comment by danmosemsft** [comment]
> is there a date(or doc to follow to know when 2.2 branch 

Not a date but it will not be many days, I expect.

**Comment by tarekgh** [comment]
@danmosemsft @maryamariyan is there more need to be done on this PR or we are just waiting for 2.2 branch to open? I am asking because I am not seeing anyone approved the changes.

**Comment by MarcoRossignoli** [comment]
@tarekgh i'm waiting 2.2 branch and that @ViktorHofer PR merge. After that i'll update my branch and re-run custom task to cleanup. I expect less code to clean up(i saw on other PR some clean-up done).

**Comment by tarekgh** [comment]
thanks @MarcoRossignoli

**Comment by tarekgh** [comment]
@MarcoRossignoli I believe you can proceed now with this PR.

@ViktorHofer  @maryamariyan if you have any more feedback on this PR, please let @MarcoRossignoli know. otherwise, please approve this PR.

File: src/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.msbuild [hunk 1]
```diff
@@ -27,7 +27,7 @@
     <CompileItem Include="$(CommonPath)\System\Net\Security\CertificateHelper.Windows.cs" />
     <CompileItem Include="$(CommonPath)\System\Runtime\ExceptionServices\ExceptionStackTrace.cs" />
     <CompileItem Include="$(CommonPath)\System\Threading\Tasks\RendezvousAwaitable.cs" />
-    <CompileItem Include="$(CommonPath)\System\Threading\Tasks\TaskToApm.cs" />
+    <CompileItem Include="$(CommonPath)\CoreLib\System\Threading\Tasks\TaskToApm.cs" />
```
Comment (MarcoRossignoli) [inline src/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.msbuild]: @maryamariyan i don't removed this line but substituted, do you agree?Now compile well. TYVM

**Comment by MarcoRossignoli** [comment]
@ViktorHofer i've merged with updates and re-run tools, now this PR should be ok, anyway if you want merge your PR before it's not a problem...after i'll resolve conflicts.

File: src/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.msbuild (hunk 1 above)
Comment (maryamariyan) [reply src/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.msbuild]: Should be fine. thanks @MarcoRossignoli.

**Comment by tarekgh** [comment]
test WIP please

**Comment by tarekgh** [comment]
@maryamariyan you may merge this if you think it is ready. Thanks.

**Pull request merged**